File: Daily Exercise/寻找旋转排序数组中的最小值.java
class Solution {
    public int finddMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            if(nums[left] < nums[right]) {
                return nums[left];
            }

            int mid = left + ((right - left) >> 1);
            if(nums[left] <= nums[mid]) {
                left = mid + 1;
            } else if(nums[left] > nums[mid]) {
                right = mid;
            }
        }
        return -1;    
    }
}

File: Daily Exercise/寻找旋转排序数组中的最小值.py
class Solution:
    def finddMin(self,nums:[]) -> int:
        
        left = 0
        right = len(nums) - 1
        while(left < right):
            mid = left + (right - left) // 2
            if nums[mid] < nums[right]:
                right = mid
            else:
                left = mid + 1
        return nums[left]

    def __init__(self):
        min = self.finddMin([8,9,10,2,5,6,7])
        print(min)


if __name__ == "__main__":
    solution = Solution()




    
